File: src/main/java/com/github/kindrat/cassandra/client/filter/Operator.java
package com.github.kindrat.cassandra.client.filter;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

import static java.util.Arrays.stream;

@Getter
@RequiredArgsConstructor
public enum Operator {
    EQ("="), GT(">"), GE(">="), LT("<"), LE("<="), NE("!="), LIKE("LIKE");

    private final String value;

    public static Operator fromValue(String value) {
        return stream(Operator.values())
                .filter(operator -> operator.getValue().equals(value)).findAny()
                .orElse(null);
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/filter/Operator.kt
package com.github.kindrat.cassandra.client.filter

import java.lang.IllegalArgumentException

enum class Operator(val value: String) {
    EQ("="), GT(">"), GE(">="), LT("<"), LE("<="), NE("!="), LIKE("LIKE");

    companion object {
        fun fromValue(value: String): Operator {
            return values().find { it.value == value } ?: throw IllegalArgumentException("Invalid operator: $value")
        }
    }
}